migration-to-python-3-only.md

```diff
@@ -0,0 +1,207 @@
+CPython core team until 2020, we propose to make the 5.x major cycle a Long
+Term Support version under the following intent:
+
+  - For the 18 month following the release of IPython 5.0, the core team will
+    engage in critical & security bug fix for the 5.x branch, as well as
+    regular release (both of minor and patch revision) of IPython 5.x branch.
+    This apply also to the documentation of the 5.x branch. THe core team will
+    also accept to provide reasonable support requests for the IPython 5.x
+    branch.
+
+
+  - For the 36 month following the release of IPython 5.x, the core team will
+    continue to review critical & security releated patches, as well as
+    improvment to the documentation, and make related release if necessary. The
+    core team will make sure that the 5.x Version of IPython stay visible
+    various web pages mentionning how to get IPython. The core team reserve the
+    right to ignore and or close any support request, request for patch
+    integration or other communication that oes not go into the previous
+    category.
+
+  - The IPython/Jupyter team will do a best effort to provide fixes in the
+    realm of reasonable requests, and will actively try to provide upstream
+    patches if IPython missbehavior is due dependedncies, unless otherwise
+    specified. I.E, we will backport and release fix to traitlets, and
+    ipykernel only if it dirrectly affect IPython
+
+  - pass 36 month after IPython 5.0 release, IPython 5.x will be considers
+    unsupported.
+
+  - The above time lines are intent, and the IPython core team reserve the
+    right to modify them in the future.
+
+  - regardless of above statement, Python 2 support in the 5.x branch will not
+    exceed support of CPython 2.7, and be likely much shorter.
+
+  - The above statements do not apply to third party who are welcome to provide
+    open or commercial support for IPython.
+
+The above timeline would lead to a release of IPython 5.0 about mid 2016,
+leading to active 5.x LTS support until about winter 2017, and passive LTS
+support until Mid 2019.
+
+## Challenges:
+
+
+### Dependencies
+
+IPython and ipykernel rely on a few dependencies maintained by the same groups
+of people, in particular: `ipykernel`, `traitlets`, and `ipython_genutils`. In
+order to provide LTS support for IPython, the IPython team, will provide
+patches for the packages requires by IPython 5.x at the condition that they
+affect the behavior of IPython. Otherwise the normal rules apply, where only
+the stable version of these software is supported, without guaranties of timeline.
+
+
+### Source distribution installation.
+
+
+The Python package manager (`pip`) does not respect the `requires-python`
+metadata field, and `setuptools` does not provide a way to set a value of this
+metadata field as far as we know. This implies that if the IPython core team release a version of
+IPython that is not compatible with IPython, the end use installing ipython
```

Comment:
I'm guessing that one of the IPythons here should be 'Python 2'. We're doing something wrong if IPython ends up incompatible with IPython ;-)